src/qt/clientmodel.cpp

```diff
@@ -99,40 +98,21 @@ size_t ClientModel::getMempoolDynamicUsage() const
     return mempool.DynamicMemoryUsage();
 }
 
-double ClientModel::getVerificationProgress() const
+double ClientModel::getVerificationProgress(const CBlockIndex *tipIn) const
 {
-    LOCK(cs_main);
-    return Checkpoints::GuessVerificationProgress(Params().Checkpoints(), chainActive.Tip());
+    CBlockIndex *tip = (CBlockIndex *)tipIn;
+    if (!tip)
+    {
+        LOCK(cs_main);
```

Comment:
This lock is only required when initially update the UI (first time, outside of the core signal scope).